### PR TITLE
ci: Update workflows towards release

### PR DESCRIPTION
**Pull request by csfreak**
This doesn't necessarily get where I wanted,  but I need to get it into main to finish working out how the release process.  

NOTE,  This will cause PRs to fail if the tittle (which is the default commit message for the squash merge) doesn't meet the conventional commit standard.

### CODE REVIEW

**Comment by csfreak** [comment]
PR Conventional Commit Validation / validate-pr-title (pull_request)  will become a required status check after merge.

**Review by stratus-ss — COMMENTED** [review]
so it seems to me that the biggest changes here are that we are implementing `go-semantic-versioning` see [here](https://github.com/s0ders/go-semver-release)

In addition the test coverage has been bumped up to 60%

There was some resizing of the test images, though the reason for this is not quite clear

Going to hold this PR until the other PR is merged so that it doesn't cause churn on the in-flight PR

**Comment by csfreak** [comment]
> There was some resizing of the test images, though the reason for this is not quite clear

the resizing of images is because the `(savefig_kwargs={"bbox_inches": "tight"})`. Which resizes the canvas to match the figure, and gets the labels in the image.  It's not perfect,  but it eliminates the cut off images.

> In addition the test coverage has been bumped up to 60%

This is just changing the requirement to better match the reality of our tests.  Eventually I would like to move this up to 95%. but we aren't there yet.

**Review by stratus-ss — APPROVED** [review]

**Pull request merged**
